balanced-optiml/bal-optiml-script/script.whizzml

```diff
@@ -0,0 +1,40 @@
+;; Requires importing the bal-optiml-library
+
+;; New project to contain the generated resources
+(define project (create-project {"name" "Balanced OptiMLs"}))
+
+;; [train-dataset validation-dataset test-dataset] collection
+;; The proportion of instances for the rare class and the rest is
+;; identical to the one in the original dataset for all of them.
+(define train-val-test-split (rare-stratified-split-fn ds-id
+                                                       training-rate
+                                                       objective-field
+                                                       rare-class
+                                                       project))
+
+
+;; List of undersampled datasets built by keeping all the instances for the
+;; rare class and randomly sampling the same amount of instances from the
+;; rest of data using different seeds. The proportion between non-rare vs. rare
+;; instaces can also be defined.
+(define undersampled-dss (undersample-fn (head train-val-test-split)
+                                         number-of-samples
+                                         objective-field
+                                         rare-class
+                                         proportion))
```

Comment:
perhaps we could check that proportion has a value within range, and number-of-samples too.  or do that in the library (i might have missed it's already done).